source/features/action-pr-link.tsx

```diff
@@ -2,25 +2,44 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '../feature-manager.js';
 import observe from '../helpers/selector-observer.js';
+import GitHubURL from '../github-helpers/github-url.js';
 
-async function add(prLink: HTMLAnchorElement): Promise<void> {
+function setSearchParameter(anchorElement: HTMLAnchorElement, name: string, value: string): void {
+	const parameters = new URLSearchParams(anchorElement.search);
+	parameters.set(name, value);
+	anchorElement.search = String(parameters);
+}
+
+async function addForRepositoryActions(prLink: HTMLAnchorElement): Promise<void> {
 	const prNumber = prLink.textContent!.slice(1);
 
 	const runLink = prLink.closest('.Box-row')!.querySelector('a.Link--primary')!;
-	const parameters = new URLSearchParams(runLink.search);
-	parameters.set('pr', prNumber);
-	runLink.search = String(parameters);
+	setSearchParameter(runLink, 'pr', prNumber);
 }
 
-async function init(signal: AbortSignal): Promise<void> {
-	observe('div.Box-row[id^=check_suite_] a[data-hovercard-type="pull_request"]', add, {signal});
+async function addForPR(actionLink: HTMLAnchorElement): Promise<void> {
+	const {branch: prNumber} = new GitHubURL(location.href);
+	setSearchParameter(actionLink, 'pr', prNumber);
+}
+
+async function initForRepositoryActionsPage(signal: AbortSignal): Promise<void> {
+	observe('div.Box-row[id^=check_suite_] a[data-hovercard-type="pull_request"]', addForRepositoryActions, {signal});
+}
+
+async function initForPRPage(signal: AbortSignal): Promise<void> {
+	observe('div.js-timeline-item details.commit-build-statuses div.merge-status-item:has(a[href="/apps/github-actions"]) a.status-actions', addForPR, {signal});
```

Comment:
This is a bit too specific. It should also work on https://github.com/refined-github/refined-github/pull/6797/commits

Reply:
FYI. Action statuses are only appeared to repository owners
But I'll implement for commits page.

I intentionally detailed before since this one.
<img width="515" alt="image" src="https://github.com/refined-github/refined-github/assets/50487467/545d92e0-9c6f-4c34-bb57-f2eb596a8058">